pkg/querier/querier.go

```diff
@@ -565,6 +565,9 @@ func (q querier) mergeSeriesSets(sets []storage.SeriesSet) storage.SeriesSet {
 
 		// SeriesSet may have some series backed up by chunks, and some not.
 		for set.Next() {
+			if ctx.Err() != nil {
+				return storage.ErrSeriesSet(ctx.Err())
+			}
```

Comment:
I think we did something similar and thanos and the perf was ok, right @yeya24 ?

Reply:
We have something like this when expanding postings. No obvious impact.